### PR TITLE
feat(bindings): add useUnhandledProps hook

### PR DESCRIPTION
**Pull request by layershifter**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds `useUnhandledProps()` hook to replace `getUnhandledProps()` function in functional components. Functionality of `useUnhandledProps()` will be changed in `compose()` PR.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12371)

### CODE REVIEW

File: packages/fluentui/react-bindings/src/hooks/useUnhandledProps.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+/**
+ * Returns an object consisting of props beyond the scope of the Component.
+ * Useful for getting and spreading unknown props from the user.
+ *
+ * @param handledProps - An array with names of props
+ * @param props - A ReactElement props object
+ * @returns A shallow copy of the prop object
+ */
+function useUnhandledProps<P extends Record<string, any>>(handledProps: (keyof P)[], props: P): Partial<P> {
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/hooks/useUnhandledProps.ts]: Can we add example in the react-bindings README? Or do you consider this as an internal hook?

File: packages/fluentui/react-bindings/src/index.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export { default as unstable_useDispatchEffect } from './hooks/useDispatchEffect
 export { default as useIsomorphicLayoutEffect } from './hooks/useIsomorphicLayoutEffect';
 export { default as useStateManager } from './hooks/useStateManager';
 export { default as useStyles } from './hooks/useStyles';
+export { default as useUnhandledProps } from './hooks/useUnhandledProps';
```
Comment (mnajdova) [inline packages/fluentui/react-bindings/src/index.ts]: As we are exporting it, I believe it deserves changelog entry

File: packages/fluentui/react-northstar/src/components/Animation/Animation.tsx [hunk 3]
```diff
@@ -231,7 +231,7 @@ const Animation: React.FC<AnimationProps> & {
   const { animationDuration, animationDelay } = animationStyles.root;
   const timeoutResult = timeout || calculateAnimationTimeout(animationDuration, animationDelay) || 0;
 
-  const unhandledProps = getUnhandledProps(Animation.handledProps, props);
+  const unhandledProps = useUnhandledProps(Animation.handledProps, props);
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Animation/Animation.tsx]: Should we use the generic typings?

Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Animation/Animation.tsx]: Nope, they will be inferred as it was previously for `getUnhandledProps`

File: packages/fluentui/react-bindings/src/index.ts (hunk 2 above)
Comment (layershifter) [reply packages/fluentui/react-bindings/src/index.ts]: Agree, added

File: packages/fluentui/react-bindings/src/hooks/useUnhandledProps.ts (hunk 1 above)
Comment (layershifter) [reply packages/fluentui/react-bindings/src/hooks/useUnhandledProps.ts]: Added 👍

**Pull request merged**